src/overlays/actors/ovl_Obj_Tree/z_obj_tree.c

```diff
@@ -71,32 +86,115 @@ static DamageTable D_80B9A59C = {
     /* Powder Keg     */ DMG_ENTRY(0, 0x0),
 };
 
-// sColChkInfoInit
-static CollisionCheckInfoInit2 D_80B9A5BC = { 8, 0, 0, 0, MASS_HEAVY };
-
-#endif
-
-extern ColliderCylinderInit D_80B9A570;
-extern DamageTable D_80B9A59C;
-extern CollisionCheckInfoInit2 D_80B9A5BC;
-
-extern UNK_TYPE D_06000680;
-extern UNK_TYPE D_06001B2C;
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Tree/ObjTree_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Tree/ObjTree_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Tree/func_80B9A20C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Tree/func_80B9A220.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Tree/func_80B9A230.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Tree/func_80B9A27C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Tree/func_80B9A348.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Tree/ObjTree_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Tree/ObjTree_Draw.s")
+static CollisionCheckInfoInit2 sColchkInfoInit = { 8, 0, 0, 0, MASS_HEAVY };
+
+extern Gfx D_06000680[];
+extern Gfx D_060007C8[];
+extern CollisionHeader D_06001B2C;
+
+void ObjTree_Init(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    ObjTree* this = THIS;
+    CollisionHeader* colHeader = NULL;
+
+    if (this->dyna.actor.params & 0x8000) {
```

Comment:
Macro for the params access. Could even point it out in the tutorial as it is something we require.